udata_gouvfr/models.py

```diff
@@ -28,3 +31,210 @@
 # Datasets
 C3 = 'c3'
 Dataset.__badges__[C3] = _('C³')
+
+
+class TerritoryDataset(object):
+    id = ''
+    title = ''
+    url_template = ''
+    description = ''
+
+    def __init__(self, territory):
+        self.territory = territory
+
+    @property
+    def url(self):
+        return self.url_template.format(code=self.territory.code)
+
+    @property
+    def slug(self):
+        return '{territory_id}-{id}'.format(
+            territory_id=self.territory.id.replace('/', '-'), id=self.id)
+
+    @property
+    def server(self):
+        return '{scheme}://{domain}'.format(
+            scheme=current_app.config['USE_SSL'] and 'https' or 'http',
+            domain=current_app.config['SERVER_NAME'])
+
+    @property
+    def api_url(self):
+        return '{server}/api/1/territories/{path}/'.format(
+            server=self.server, path=self.slug)
+
+
+class ResourceBasedTerritoryDataset(TerritoryDataset):
+    dataset_id = ''
+    resource_id = ''
+    territory_attr = ''
+    csv_column = ''
+
+    @property
+    def url(self):
+        return '{url}?{parameters}'.format(
```

Comment:
No need to anually concatenate the parameters, Flask automatically add (and encode) them if not defined in the url pattern:
```python
return url_for('gouvfr.territory_dataset_resource', territory=self.territory,
                                                    dataset=self.dataset_id,
                                                    resource_id=self.resource_id,
                                                    territory_attr=self.territory_attr,
                                                    csv_column=self.csv_column)
```